core/common/src/main/java/alluxio/underfs/UnderFileSystem.java

```diff
@@ -338,6 +337,13 @@ public int getValue() {
   @Deprecated
   long getBlockSizeByte(String path) throws IOException;
 
+  /**
+   * Gets the under file system configuration.
+   *
+   * @return the configuration
+   */
+  AlluxioConfiguration getConfiguration() throws IOException;
```

Comment:
This is a public API, so ideally we provide a default implementation in the interface for back-compat with something like  
```
  default AlluxioConfiguration getConfiguration() {
    return <empty AlluxioConfiguration>;
  }
```
As far as I remember, the reason for this being a public API is ufs extensions. If we change it, any ufs extension not deriving from `BaseUfs` has to implement this new method (other than re-compilation for new Alluxio version). This may or may not be an issue since I can't think of any known ufs extension which does not.

Reply:
fixed